[docs] Tutorial: Remove web references

amandeepmittal opened the pull request:
# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

Fixes ENG-5249 & #17824 

# How

<!--
How did you build this feature or fix this bug and why?
-->

This PR removes references of the "web" part used in the Tutorial. It hides the "Handling Platform differences" page and updates the inline Snack example on "Sharing" page.

Additionally, all internal links are converted to use the full path and verified any external links to make sure they are working.

_Note:_ This is a workaround for the web part for now.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->
The changes have been tested by running docs locally.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

amandeepmittal committed:
[docs] Tutorial: Remove web references

amandeepmittal committed:
Remove extra space

brentvatne commented:
i think instead of removing web entirely we could add some platform check to say this isn't supported on web and to check out the ios / android app - this way we at least acknowledge that web exists and people can run their app in it with some reasonable output

kbrandwijk commented:
> i think instead of removing web entirely we could add some platform check to say this isn't supported on web and to check out the ios / android app - this way we at least acknowledge that web exists and people can run their app in it with some reasonable output

Keeping in mind this is a temporary change until we have a proper solution for web, making additional changes the the sample to continue to work on web for now doesn't seem worth it. I'll let @jonsamp chime in though, as he has this discussed this direction with @amandeepmittal for this PR.

brentvatne commented:
i'm fine with whatever is decided, but i do think it wouldn't take more than 20-30 mins to make the change i suggested, and we don't know when the new tutorial will be done and don't want to rush it

amandeepmittal commented:
> i'm fine with whatever is decided, but i do think it wouldn't take more than 20-30 mins to make the change i suggested, and we don't know when the new tutorial will be done and don't want to rush it

I can revert back some of the changes around "web" part and add the platform check.

amandeepmittal committed:
Revert back some of the web references to acknowledge its existence

amandeepmittal committed:
Change the title

amandeepmittal commented:
@brentvatne I have reverted back some of the references for "web" part through out the turorial. The platform check already existed in the Snack Example.

To reuse the platform difference example from Snack on "Sharing" page, I added a new section title called "Platform differences with web" that explains why Share API doesn't work on web. For this, I am re-using some of the content from the "Handling Platform differences" docs page. I think it was pretty well explained on why Web Share API doesn't work in Chrome or a web browser on the desktop.

This hopefully will address that Expo tools are compatible on all platforms with some exceptions.

jonsamp reviewed (approved):
Thanks for removing this content + adding a section where we check if the platform is web. As we work on a newer/better tutorial, this is a good step for the current tutorial.
